### DISCUSSION TITLE
Is there a way to hide the credentials of the URL in the mkdocs.yml?

### DISCUSSION THREAD

**Discussion by alexleong4438** [mkdocs/mkdocs#3716] [Q&A]
I'm going to attach an external URL link in my mkdocs.yml (nav section). However, the external URL requires credentials to log in. For example, in the **mkdocs.yml**:

> nav:
          - Introduction: 'index.md'
          - API reference: https://**username:password**@<my_website_url>

Is there a way to hide the **username:password** after building the static web pages so that the user won't be aware of it?

**Comment by Andre601** [comment]
At most could you use the `!env` feature.
HOWEVER, this would still leave the cridentials in the url available in the built page.

Why do you even need cridentials for an API reference?

**Comment by pawamoy** [comment]
Since `my_website_url` is an external page, its contents are not included in the final site. The navigation will simply point at it. Therefore, no, it is not possible to hide the credentials from your readers, since they will *need* the credentials to access the page.

You could try to rely on other authentication methods, for example a cookie they'd have in the browser session, set by your company workstations or something. If this is for the public, then remove authentication :shrug:
